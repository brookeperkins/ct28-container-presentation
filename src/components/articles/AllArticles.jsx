import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const AllArticles = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={article.title}>
      <Article {...article} />
    </li>
  ));

  return (
    <ul data-testid="articles">
      {articleElements}
    </ul>
  );
};

AllArticles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })).isRequired
};

export default AllArticles;
