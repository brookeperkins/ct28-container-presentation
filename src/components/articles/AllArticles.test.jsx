import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AllArticles from './AllArticles';
/* eslint-disable max-len */

const newsArray = [
  { title: 'This Title', author: 'Lil Wayne', description: 'What a fabulous article.' },
  { title: 'That Title', author: 'Margaret Atwood', description: 'Stupendous article.' },
  { title: 'Those Titles', author: 'Norman Mailer', description: 'Eh.' }
];

describe('AllArticles component', () => {
  afterEach(() => cleanup());
  it('render AllArticles component', () => {
    const { asFragment } = render(<AllArticles
      articles={newsArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
