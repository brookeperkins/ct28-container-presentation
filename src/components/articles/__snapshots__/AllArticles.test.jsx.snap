// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllArticles component render AllArticles component 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <li>
      <h3>
        This Title
      </h3>
      <p>
        By: Lil Wayne
      </p>
      <p>
        Summary: What a fabulous article.
      </p>
    </li>
    <li>
      <h3>
        That Title
      </h3>
      <p>
        By: Margaret Atwood
      </p>
      <p>
        Summary: Stupendous article.
      </p>
    </li>
    <li>
      <h3>
        Those Titles
      </h3>
      <p>
        By: Norman Mailer
      </p>
      <p>
        Summary: Eh.
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
